### ISSUE TITLE
find_essential fails to recover the pose for about half of all samples: clamp_min(1e-8) destroys a negative leading coefficient (regression from #3458)

### ISSUE THREAD

**Issue by VanniLeonardo** [kornia/kornia#4847]
### What happened?

`find_essential` fails to recover the pose for about half of all ordinary, noise-free 5-point samples. The cause is one line in `_null_to_Nister_solution_script` (`kornia/geometry/epipolar/essential.py`), where the companion matrix of the degree-10 polynomial is normalized by its leading coefficient:

```python
cs_de = cs[:, -1].clamp_min(1e-8)
C[:, -1, :] = -cs[:, :-1] / cs_de.unsqueeze(-1)
```

`clamp_min(1e-8)` guards against a zero leading coefficient, but it also replaces every **negative** leading coefficient with `+1e-8`. That flips the sign of the companion row and scales it by `1 / |cs[-1]|`, so the eigenvalues are no longer the polynomial's roots and none of the 10 candidates is the true essential matrix. The leading coefficient is negative for about half of all samples.

This is a regression from #3458 (`e9e93584`, 2026-01-06). Before it the guard replaced only an exact zero:

```python
cs_de = cs[:, -1].unsqueeze(-1)
cs_de = torch.where(cs_de == 0, torch.tensor(1e-8, dtype=cs_de.dtype), cs_de)
```

Measured on 2000 random noise-free scenes (random rotation up to about 11 degrees per axis, unit translation, 5 points at depth 3 to 7), counting a pose as recovered when the best of the 10 candidates is within the tolerance of the ground-truth `E`, up to sign:

| | negative leading coefficient | recovered on `main` | recovered with the pre-#3458 guard |
|---|---|---|---|
| float64, error below 1e-8 | 1017 / 2000 | 992 / 2000 (17 of the 1017) | 1979 / 2000 |
| float32, error below 1e-3 | 1008 / 2000 | 880 / 2000 (6 of the 1008) | 1767 / 2000 |

The median best-candidate error at float64 goes from 2.75e-08 to 2.42e-14. Samples with a positive leading coefficient give bitwise identical output with either guard.

Which particular samples fail depends on the platform, because the sign of the coefficient follows the sign of the null-space basis the SVD returns. That explains a discrepancy in the #4766 review. The noise-free pose sample posted there recovered `E` to 7.1e-16 on macOS arm64 but only to 1.756e-4 at float64 and 0.8604 at float32 on Linux x86. Here its leading coefficient is -3.1e-4. With the pre-#3458 guard it recovers `E` to 9.1e-15 at float64 and 1.05e-06 at float32.

Nothing caught it because `test_synthetic_sampson` uses one sample whose coefficient is positive on the platforms that ran it, and `RANSAC(model_type="essential")` hides it by drawing more minimal samples. RANSAC still needs roughly twice as many to find the model, and a differentiable pipeline that calls `find_essential` on one sample gets a wrong pose half the time.

### Minimal reproduction

```python
import torch
import kornia.geometry.epipolar as epi
from kornia.geometry.conversions import axis_angle_to_rotation_matrix

torch.manual_seed(0)
n, dtype = 500, torch.float64
X1 = torch.rand(n, 5, 3, dtype=dtype) * torch.tensor([2.0, 2.0, 4.0], dtype=dtype) + torch.tensor([-1.0, -1.0, 3.0], dtype=dtype)
R = axis_angle_to_rotation_matrix((torch.rand(n, 3, dtype=dtype) - 0.5) * 0.4)
t = torch.nn.functional.normalize(torch.rand(n, 3, dtype=dtype) - 0.5, dim=-1)
X2 = X1 @ R.transpose(-1, -2) + t[:, None]
x1, x2 = X1[..., :2] / X1[..., 2:], X2[..., :2] / X2[..., 2:]          # noise-free correspondences
E_gt = epi.essential_from_Rt(torch.eye(3, dtype=dtype).expand(n, 3, 3), torch.zeros(n, 3, 1, dtype=dtype), R, t[..., None])

E = epi.find_essential(x1, x2, torch.ones(n, 5, dtype=dtype))       # (n, 10, 3, 3)
unit = lambda M: M / M.flatten(-2).norm(dim=-1)[..., None, None]
err = torch.minimum((unit(E) - unit(E_gt)[:, None]).flatten(-2).norm(dim=-1),
                    (unit(E) + unit(E_gt)[:, None]).flatten(-2).norm(dim=-1)).nan_to_num(1.0).min(-1).values
print(f"pose recovered to 1e-8: {int((err < 1e-8).sum())} / {n}, median error {err.median().item():.2e}")
```

Observed on `main`:

```
pose recovered to 1e-8: 255 / 500, median error 3.45e-10
```

With the pre-#3458 guard restored:

```
pose recovered to 1e-8: 492 / 500, median error 1.90e-14
```

### Environment

- Kornia version: 0.9.0rc1 (current `main`, `9c39a109`)
- PyTorch version: 2.9.1+cu128
- Python version: 3.11.14
- OS / device: Linux x86, CPU

### Anything else?

The fix is to restore the pre-#3458 semantics, replacing only an exact zero, with a device-safe `torch.full_like` in place of `torch.tensor`. I have it ready with a platform-independent test and will open a PR.

The exactly zero case is not changed by that fix. A degenerate sample with zero parallax, for example `p1 == p2`, can make the leading coefficient exactly 0. Normalizing by `1e-8` then gives a companion row around 1e37, and LAPACK returns a repeated zero eigenvalue with a singular eigenvector matrix. That is the cause of the backward failures in #4831, and it is left there.

Investigated with Claude Code. Every figure above is from an executed run on `main` `9c39a109`.
